### PR TITLE
more tests for persistence

### PR DESCRIPTION
**Pull request by mfromme**

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/datadonation/common/persistence/domain/AndroidId.java [hunk 1]
```diff
@@ -7,6 +7,10 @@
 @Table("android_id")
 public class AndroidId {
 
+  public AndroidId(String id) {
+    setId(id);
+  }
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/datadonation/common/persistence/domain/AndroidId.java:12]: by declaring this constructor, the default (empty) constructor is gone... does everything still work? All the DB magic?

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java [hunk 2]
```diff
@@ -29,7 +30,7 @@ final class ProdSrsRateLimitVerificationStrategyTest {
 
   @Test
   void testInvalidSrsRateLimit() {
-    final AndroidId androidId = new AndroidId();
+    final AndroidId androidId = new AndroidId(UUID.randomUUID().toString());
```
Comment (hilmarf) [inline services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java:33]: can we not just use the non-args constructor and then use setId?
```suggestion
    final AndroidId androidId = new AndroidId();
    androidId.setId(UUID.randomUUID().toString());
```

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java [hunk 3]
```diff
@@ -39,7 +40,7 @@ void testInvalidSrsRateLimit() {
 
   @Test
   void testValidSrsRateLimit() {
-    final AndroidId androidId = new AndroidId();
+    final AndroidId androidId = new AndroidId(UUID.randomUUID().toString());
```
Comment (hilmarf) [inline services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java:43]: see above

**Review by hilmarf — CHANGES_REQUESTED** [review]
please don't introduce constructors for testing purpose only

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java (hunk 2 above)
Comment (mfromme) [reply services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java:33]: I took this over from the way OneTimePassword works. I can't say 100% if this affects anything. But all tests still work.

File: services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java (hunk 3 above)
Comment (mfromme) [reply services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/attestation/ProdSrsRateLimitVerificationStrategyTest.java:43]: see above :-)

**Comment by mfromme** [comment]
Should I revert the constructor changes? It was purely cosmetic and also because another class does it the same way...

**Review by hilmarf — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Pull request merged**